lib/carrierwave/storage/aws_file.rb

```diff
@@ -37,7 +37,12 @@ def filename(options = {})
       end
 
       def read
-        file.get(aws_options.read_options).body.read
+        if block_given?
+          file.get(aws_options.read_options) { |chunk| yield chunk }
```

Comment:
This looks good, but it would be nice to assign `aws_options.read_options` within the method rather than calling into it twice, i.e. `read_options = aws_options.read_options`